MarsQACompetitionTaskNUnit/Pages/EducationPage.cs

```diff
@@ -0,0 +1,326 @@
+                WaitUtils.WaitToBeVisible(driver, "Xpath", "getUniversity", 60);
+                getUniversity = driver.FindElement(By.XPath($"//div[@data-tab='third']//table/tbody[{rowNumber}]/tr/td[2]")).Text;
+                getTitle = driver.FindElement(By.XPath($"//div[@data-tab='third']//table/tbody[{rowNumber}]/tr/td[3]")).Text;
+                getDegree = driver.FindElement(By.XPath($"//div[@data-tab='third']//table/tbody[{rowNumber}]/tr/td[4]")).Text;
+                getGraduationYear = driver.FindElement(By.XPath($"//div[@data-tab='third']//table/tbody[{rowNumber}]/tr/td[5]")).Text;
+
+                ReportLogger.LogInfo($"Retrieved [row {rowNumber}]: Country: {getCountry}, University: {getUniversity}");
+                if (Country.Equals(getCountry) && University.Equals(getUniversity) && Title.Equals(getTitle) && Degree.Equals(getDegree) && GraduationYear.Equals(getGraduationYear)) { recordPresent = true; }
+            }
+
+            return recordPresent;
+        }
+
+        public int RowCount()
+        {
+            return EducationRows.Count;
+        }
+
+        public int GetEducationRow(string University, string Country, string Title, string Degree, string GraduationYear)
+        {
+            string getUniversity, getCountry, getTitle, getDegree, getGraduationYear;
+
+            for (int i = 1; i <= RowCount(); i++)
+            {
+                try
+                {
+                    getCountry = driver.FindElement(By.XPath($"//div[@data-tab='third']//table/tbody[{i}]/tr/td[1]")).Text;
+                    getUniversity = driver.FindElement(By.XPath($"//div[@data-tab='third']//table/tbody[{i}]/tr/td[2]")).Text;
+                    getTitle = driver.FindElement(By.XPath($"//div[@data-tab='third']//table/tbody[{i}]/tr/td[3]")).Text;
+                    getDegree = driver.FindElement(By.XPath($"//div[@data-tab='third']//table/tbody[{i}]/tr/td[4]")).Text;
+                    getGraduationYear = driver.FindElement(By.XPath($"//div[@data-tab='third']//table/tbody[{i}]/tr/td[5]")).Text;
+                    if (Country.Equals(getCountry) && University.Equals(getUniversity) && Title.Equals(getTitle) && Degree.Equals(getDegree) && GraduationYear.Equals(getGraduationYear))
+                    {
+                        ReportLogger.LogInfo($"Education Present at row: {i}");
+                        return i;
+                    }
+                }
+                catch (NoSuchElementException)
+                {
+                    continue;
+                }
+            }
+            return 0;
+        }
+
+        public void EditEducationRecord(string University, string Country, string Title, string Degree, string GraduationYear)
+        {
+
+            // click edit pencil icon for the existing record
+            WaitUtils.WaitToBeVisible(driver, "Xpath", "LastEditPencilIcon", 10);
+            LastEditPencilIcon.Click();
+
+            if (University.Length > 0)
```

Comment:
why are you checking if length is greater?

Reply:
same comments apply for the below

Reply:
updated as per comment